### PR TITLE
Energy optimized and manual check warning dialogs (EXPOSUREAPP-1853)

### PR DESCRIPTION
**Pull request by Oliver-Zimmerman**
## Checklist

* [x] Test your changes as thoroughly as possible before you commit them. Preferably, automate your test by unit/integration tests.
* [ ] If this PR comes from a fork, please [Allow edits from maintainers](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork)
* [x] Set a speaking title. Format: {task_name} (closes #{issue_number}). For example: Use logger (closes # 41)
* [ ] [Link your Pull Request to an issue](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) - Pull Requests that are not linked to an issue with you as an assignee will be closed, except for minor fixes for typos or grammar mistakes in *documentation or code comments*.
* [ ] Create Work In Progress [WIP] pull requests only if you need clarification or an explicit review before you can continue your work item.
* [x] Make sure that your PR is not introducing _unnecessary_ reformatting (e.g., introduced by on-save hooks in your IDE)
* [ ] Make sure that your PR does not contain changes in text files, therefore the PR must not contain changes in values/strings/* and / or assets/* (see issue #332 for further information)
* [x] Make sure that your PR does not contain compiled sources (already set by the default .gitignore) and / or binary files

## Description
- Created a Dialog in MainFragment that shows once (shared pref boolean check), and if energy-optimized mode is enabled
- Created a shared pref in local data to store if the energy -optimized warning dialog has been shown
- Added strings required for dialog (Only DE confirmed)
- Adjusted existing dialog in onboarding process
- Added second dialog explaining manual checks, this is shown during onboarding and in SettingsTracingFragment.kt when tracing is enabled when background is not prioritized. 

Closes JIRA task EXPOSUREAPP-1853

### CODE REVIEW

**Review by janetback — DISMISSED** [review]
Checked text strings.

File: Corona-Warn-App/src/main/res/values-de/strings.xml [hunk 1]
```diff
@@ -135,7 +135,7 @@
     <string name="risk_card_body_contact">"Bisher keine Risiko-Begegnungen"</string>
     <!-- XTXT: risk card - number of contacts for one or more -->
     <plurals name="risk_card_body_contact_value">
-        <item quantity="one">"%1$s Risiko-Begegnung"</item>
+        <item quantity="one">"%1$s Risiko-Begegnungen mit niedrigem Risiko"</item>
```
Comment (MichaelPFrey) [inline Corona-Warn-App/src/main/res/values-de/strings.xml:138]: This looks like an artifact which does not belong in this PR.  see also #937

File: Corona-Warn-App/src/main/res/values-de/strings.xml [hunk 2]
```diff
@@ -135,7 +135,7 @@
     <string name="risk_card_body_contact">"Bisher keine Risiko-Begegnungen"</string>
     <!-- XTXT: risk card - number of contacts for one or more -->
     <plurals name="risk_card_body_contact_value">
-        <item quantity="one">"%1$s Risiko-Begegnung"</item>
+        <item quantity="one">"%1$s Risiko-Begegnungen"</item>
```
Comment (vaubaehn) [inline Corona-Warn-App/src/main/res/values-de/strings.xml:138]: Should be singular here, like in current dev:
Risiko-Begegnung

**Review by vaubaehn — CHANGES_REQUESTED** [review]
String in line 138 should be reverted back to singular.

**Review by pwoessner — APPROVED** [review]

**Pull request merged**